warehouse/integrations/vulnerabilities/utils.py

```diff
@@ -101,19 +101,26 @@ def _analyze_vulnerability(request, vulnerability_report, origin, metrics):
         )
         raise
 
+    found_releases = False  # by now, we don't have any release found
+
     for version in report.versions:
         try:
             release = _get_release(request, project, version)
+            found_releases = True  # at least one release found
         except NoResultFound:
             metrics.increment(
                 "warehouse.vulnerabilities.error.release_not_found",
                 tags=[f"origin:{origin}"],
             )
-            raise
+            continue  # skip that release
 
         if release not in vulnerability_record.releases:
             vulnerability_record.releases.append(release)
 
+    if not found_releases:
+        # no releases found, then raise an exception
+        raise NoResultFound("None of the releases were found")
```

Comment:
This is a SQLAlchemy exception, not one we want to manually raise. We should just raise a generic exception here for the time being.

I think we should also add a test for this, the existing test doesn't assert on this behavior in any way.

Reply:
> This is a SQLAlchemy exception, not one we want to manually raise. We should just raise a generic exception here for the time being.

I was thinking about `KeyError`, or `ValueError`. Is that right?

Reply:
No, I don't think so. Since this will eventually become a HTTP response, I think `pyramid.httpexceptions. HTTPBadRequest` would be best.